data_steward/cdr_cleaner/cleaning_rules/remove_non_matching_participant.py

```diff
@@ -31,218 +38,247 @@
+        job = client.query(create_sandbox_not_match_person_id)
+        job.result()
+
+        for hpo_id in readers.get_hpo_site_names():
+
+            identity_match = resources.get_table_id(IDENTITY_MATCH,
+                                                    hpo_id=hpo_id)
+
+            if not self.exist_table(client, self.project_id,
+                                    self.validation_dataset_id, identity_match):
+                LOGGER.info(
+                    f'{hpo_id} is missing the identity_match table. Skipping...'
+                )
+                continue
+
+            participant_match = resources.get_table_id(PARTICIPANT_MATCH,
+                                                       hpo_id=hpo_id)
+
+            if self.exist_table(client, self.project_id, self.ehr_dataset_id,
+                                participant_match):
+                only_not_validated_id_condition = ONLY_NOT_VALIDATED_ID_CONDITION.render(
+                    project_id=self.project_id,
+                    ehr_dataset_id=self.ehr_dataset_id,
+                    participant_match_table=participant_match)
+            else:
+                only_not_validated_id_condition = ''
+                LOGGER.info(
+                    f'{hpo_id} is missing the participant_match table. '
+                    f'All data from {hpo_id} is treated as not validated.')
+
+            key_fields_criteria, all_fields_criteria = self.get_fields_criteria(
+            )
+
+            not_validated_participants_query = INSERT_NOT_MATCH_PERSON_ID.render(
+                project_id=self.project_id,
+                sandbox_dataset_id=self.sandbox_dataset_id,
+                not_match_person_id_table=self.sandbox_table_for(NOT_MATCH),
+                participant_match_table=participant_match,
+                key_fields_criteria=key_fields_criteria,
+                all_fields_criteria=all_fields_criteria,
+                validation_dataset_id=self.validation_dataset_id,
+                identity_match_table=identity_match,
+                only_not_validated_id_condition=only_not_validated_id_condition)
+
+            job = client.query(not_validated_participants_query)
+            job.result()
+
+    def setup_validation(self, client: BigQueryClient) -> None:
+        pass
+
+    def validate_rule(self, client: BigQueryClient) -> None:
+        pass
+
+    def get_sandbox_tablenames(self) -> list:
+        """
+        Return a list table names created to backup deleted data.
+
+        NOTE: self.sandbox_table_for(NOT_MATCH) is not included in this list
+        because it is a temporary lookup table, not a backup table.
+        """
+        return [
+            self.sandbox_table_for(table)
+            for table in remove_pids.get_tables_with_person_id(
+                self.project_id, self.dataset_id)
+        ]
+
+    def exist_table(self, client: BigQueryClient, project_id, dataset_id,
```

Comment:
We can remove this function when the following JIRA ticket is implemented. Shall we wait for that?
https://precisionmedicineinitiative.atlassian.net/browse/DC-2460